develop/TOOLS/NuGet/Run_Time_Assembly_Binding.md

```diff
@@ -4,6 +4,36 @@ uid: Run_Time_Assembly_Binding
 
 # Run-time assembly binding
 
+## Introduction
+
+At run-time, assemblies are loaded the moment they are needed by the executing code. When the Just-In-Time (JIT) compiler compiles the intermediate language (IL) to native code, it sees which types are referenced. The JIT compiler will then determine the assemblies that define these types, and these will be loaded.
+
+The .NET Framework follows specific steps while trying to locate an assembly at run-time. For more information about the internals of assembly loading in the .NET Framework, refer to [How the Runtime Locates Assemblies](https://learn.microsoft.com/en-us/dotnet/framework/deployment/how-the-runtime-locates-assemblies). Custom paths to be searched for while locating an assembly can be specified via the [AssemblyResolve](https://learn.microsoft.com/en-us/dotnet/api/system.appdomain.assemblyresolve?view=netframework-4.8) event. This event is triggered on an AppDomain level and the event provides information about the assembly that the runtime is trying to resolve. This event is triggered when the run-time could not find the assembly in any of the previous steps the runtime executes during assembly resolving.
+
+In DataMiner, for SLScripting, this event is implemented in such a way that it will search in the `C:\Skyline DataMiner\ProtocolScripts`, `C:\Skyline DataMiner\Files` and `C:\Skyline DataMiner\ProtocolScripts/DllImport` folders and also in any of the folders denoted by the [dllImport](xref:Protocol.QActions.QAction-dllImport) attribute of a QAction (or the [Param type="ref"](xref:DMSScript.Script.Exe.Param-type) tags in Automation scripts).
```

Comment:
Maybe specify the order as well? DllImport path has forward slash as well.

Reply:
added more explanation with order